plugins/modules/monitoring/pagerduty_change.py

```diff
@@ -0,0 +1,189 @@
+    required: false
+    type: str
+  environment:
+    description:
+      - The environment name, typically 'production', 'staging', etc.
+    required: false
+    type: str
+  link_url:
+    description:
+      - A URL where more information about the deployment can be obtained.
+    required: false
+    type: str
+  link_text:
+    description:
+      - Descriptive text for a URL where more information about the deployment can be obtained.
+    required: false
+    type: str
+  url:
+    description:
+      - URL to submit the change event to.
+    required: false
+    default: https://events.pagerduty.com/v2/change/enqueue
+    type: str
+  validate_certs:
+    description:
+      - If C(no), SSL certificates for the target URL will not be validated.
+        This should only be used on personally controlled sites using self-signed certificates.
+    required: false
+    default: yes
+    type: bool
+notes: Supports C(check_mode).
+'''
+
+EXAMPLES = '''
+- name: Track the deployment as a PagerDuty change event
+  community.general.pagerduty_change:
+    integration_key: abc123abc123abc123abc123abc123ab
+    summary: The application was deployed
+
+- name: Track the deployment as a PagerDuty change event with more details
+  community.general.pagerduty_change:
+    integration_key: abc123abc123abc123abc123abc123ab
+    summary: The application was deployed
+    source: Ansible Deploy
+    user: ansible
+    repo: github.com/ansible/ansible
+    revision: '4.2'
+    environment: production
+    link_url: https://github.com/ansible-collections/community.general/pull/1269
+    link_text: View changes on GitHub
+'''
+
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+from datetime import datetime
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            integration_key=dict(required=True, type='str'),
+            summary=dict(required=True, type='str'),
+            source=dict(required=False, default='Ansible', type='str'),
+            user=dict(required=False, type='str'),
+            repo=dict(required=False, type='str'),
+            revision=dict(required=False, type='str'),
+            environment=dict(required=False, type='str'),
+            link_url=dict(required=False, type='str'),
+            link_text=dict(required=False, type='str'),
+            url=dict(required=False,
+                     default='https://events.pagerduty.com/v2/change/enqueue', type='str'),
+            validate_certs=dict(default=True, type='bool')
+        ),
+        supports_check_mode=True
+    )
+
+    # API documented at https://developer.pagerduty.com/docs/events-api-v2/send-change-events/
+
+    url = module.params['url']
+    headers = {'Content-Type': 'application/json'}
+
+    if module.check_mode:
+        _response, info = fetch_url(
+            module, url, headers=headers, method='POST')
+
+        if info['status'] == 400:
```

Comment:
This looks strange. 400 is a Bad Request error. Also this always results in `changed=True` or a module failure, which looks wrong for a regular Ansible module (keyword: idempotency).

Reply:
Right. This is my first time building something like this so I took a lot of ... insipration ... from other modules that do similar things. I original had check mode doing something similar to this: https://github.com/ansible-collections/community.general/blob/main/plugins/modules/monitoring/honeybadger_deployment.py#L106

I got some feedback that it might be good to at least try to verify that the API/URL were correct so I added this. It essentially tries to test that the API call is configured properly but doesn't pass any data and thus is a bad request.

Do you have a suggestion for a better way?

Reply:
Usually modules operate the following way:
1. (Try to) get state about object from remote API;
2. If object exists, compare object to parameters. If they differ, update the object.
3. If object does not exist, create it.

When check mode is active, the module simply does not do the actual changes (update resp. creation).

Of course it depends on the API, whether querying for existence is possible at all, or whether updating is possible at all - sometimes objects must be destroyed and re-created, or there's nothing that can be done except failing. But failing should be the last resort.